tests/main/layout-change/task.yaml

```diff
@@ -0,0 +1,43 @@
+summary: Ensure that replacing a bind with a symlink doesn't break an update
+
+details: |
+    Recently, the steam snap replaced several bindings with symlinks in the
+    layout part, and that resulted in users not being able to update it from
+    the "binded" version to the "symlinked" one because snapd complained that
+    it couldn't update the mount namespace. The solution consists on discard
+    the namespace and retry again.
+
+systems:
+    # the snaps aren't available for ARM or 32bit systems.
+    - -ubuntu-18.04-32
+    - -ubuntu-20.04-arm-64
+
+prepare: |
+    echo "Removing snaps"
+    snap remove --purge test-snapd-layout-change || true
```

Comment:
Silly question, why is it removing the snaps in prepare? In what situation can they be installed when prepare runs?

Reply:
Well... just in case, to ensure that there is no trash from failed executions...

Reply:
(maybe I'm too much cautious)

Reply:
It shouldn't really be needed but it will do no harm